### PR TITLE
tests: lib: c_lib: fix double promotion error

### PR DESCRIPTION
**Pull request by XenuIsWatching**
Fix double promotion warnings

see: https://github.com/zephyrproject-rtos/zephyr/pull/57154

### CODE REVIEW

**Review by fabiobaltieri — DISMISSED** [review]

**Review by cfriedt — DISMISSED** [review]

**Comment by XenuIsWatching** [comment]
Please don't merge yet. There's still another issue that I need to fix.

**Comment by XenuIsWatching** [comment]
fixed now... apparently that warning flag uncovered a typo with the defines for NAN and NANF which were swapped >_<

**Comment by XenuIsWatching** [comment]
Gonna hold off on merging for now. There's gonna be some deep investigations that have been uncovered...

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by XenuIsWatching** [comment]
> Gonna hold off on merging for now. There's gonna be some deeper investigations that have been uncovered...

everything looks good now ;)

**Review by cfriedt — DISMISSED** [review]

File: tests/lib/c_lib/common/src/test_sqrt.c [hunk 1]
```diff
@@ -13,11 +13,11 @@
 #define local_abs(x) (((x) < 0) ? -(x) : (x))
 
 #ifndef NAN
-#define NAN	(__builtin_nansf(""))
+#define NAN	(__builtin_nans(""))
 #endif
 
 #ifndef NANF
```
Comment (keith-packard) [inline tests/lib/c_lib/common/src/test_sqrt.c:19]: This code probably shouldn't use NANF -- the C standard (at least C99, C11 and C17) only defines NAN:

> The macro
>
> NAN
>
> is defined if and only if the implementation supports quiet NaNs for the float type. It
> expands to a constant expression of type float representing a quiet NaN.

Similarly, the only defined constant for infinity is `INFINITY`, also of `float` type.

Comment (XenuIsWatching) [reply tests/lib/c_lib/common/src/test_sqrt.c:19]: ok... well then... I'll guess I'll just remove `NANF` and `INFF` and cast them to floats 🤷 in the sqrtf test

Comment (XenuIsWatching) [reply tests/lib/c_lib/common/src/test_sqrt.c:19]: done... also, renamed `INF` to `INFINITY`

**Review by keith-packard — APPROVED** [review]
Looks awesome. Thanks for doing this!

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**